osquery/tables/system/windows/registry.cpp

```diff
@@ -151,14 +166,16 @@ void queryKey(const std::string& keyPath, QueryData& results) {
     return;
   }
 
-  BYTE* bpDataBuff = new BYTE[cbMaxValueData];
   DWORD cchValue = maxKeyLength;
   TCHAR achValue[maxValueName];
 
   // Process registry values
   for (size_t i = 0, retCode = ERROR_SUCCESS; i < cValues; i++) {
+    DWORD lpData = (cbMaxValueData == 0) ? NULL : cbMaxValueData;
+    DWORD lpType;
+    BYTE* bpDataBuff =
+        (cbMaxValueData == 0) ? NULL : new BYTE[cbMaxValueData]();
     size_t cnt = 0;
-    ZeroMemory(bpDataBuff, cbMaxValueData);
```

Comment:
Not that this was particularly necessary in the first place, but any reason it was removed?